platformio/package/meta.py

```
@@ -423,8 +423,21 @@ def _parse_name_from_uri(uri):
             if c in uri:
                 uri = uri[: uri.index(c)]
 
-        # parse real repository name from Github
         parts = urlparse(uri)
+
+        # Handle browse URLs with subdirectory
+        # GitHub/GitLab: /user/repo/tree/branch/subdir
+        if parts.netloc in ("github.com", "gitlab.com"):
+        path_parts = [p for p in parts.path.split("/") if p and p != "-"]
```

Pylint (4.0.4)

[error] Parsing failed: 'expected an indented block after 'if' statement on line 430 (platformio.package.meta, line 431)'

(E0001)